src/main/java/com/thealgorithms/backtracking/HamiltonianCycle.java

```diff
@@ -0,0 +1,149 @@
+package com.thealgorithms.backtracking;

```

Comment:
Add comments about the algorithm, and put some links where people can read and know about the algorithm